tutorials/tech-terms/index.md

```diff
@@ -0,0 +1,275 @@
+
+<ExpansionPanel title="Why is it called light client daemon?">
+
+Before SDK v0.40, to get a REST API it was necessary to run another backend service (or "[daemon](https://en.wikipedia.org/wiki/Daemon_(computing))", a label inherited from Unix), for example using `gaiacli rest-server --laddr 0.0.0.0:1317 --node localhost:26657`. In Cosmos SDK v0.40, REST was moved inside the node service making it a tool of the Cosmos SDK, but the "daemon" element stuck, leading to the name _light client daemon_ (LDC).
+
+</ExpansionPanel>
+
+## Remote procedure call (RPC)
+
+A **remote procedure call (RPC)** is _a software communication protocol_. The term is often found in distributed computing because RPC is a technique to realize inter-process communication (IPC) by allowing a program to cause a subroutine procedure that is executed in a different address space (a different machine).
+
+RPC can be understood as a client-server interaction in which the "caller" is the client, more specifically the requesting program, and the "executor" is the server, more specifically the service-providing program. The interaction is implemented through a request-response message-passing system.
+
+In short, RPC is a request-response protocol, initiated by a client sending a request to a remote server to execute the subroutine. 
+
+Usually, the called functions are run on a different computer than the one calling them. However, with RPC, the developer codes as if the subroutine would be local; the developer does not have to code in details for remote interaction. Thus, with RPCs it is implied that all calling procedures are basically the same, independent of them being local or remote calls.
+
+<HighlightBox type="note">
+
+As RPCs implement remote request-response protocols, it is important to note that remote procedure calls can fail in case of network problems.
+
+</HighlightBox>
+
+### How does an RPC request work?
+
+In general, when a remote procedure call is invoked the procedure parameters are transferred across the network to the execution environment where the procedure is executed. Once finished, the results of the procedure call invoked are transferred to the call environment. The execution then resumes in the call environment, just as it would in a regular local procedure call.
+
+A step-by-step RPC request could look like the following:
+
+1. A client calls a client stub - a piece of code converting parameters that are passed between client and servers during an RPC. The call is a local procedure call.
+
+<HighlightBox type="info">
+
+A stub is a small program routine substituting a longer program.
+
+In RPCs, the client's stub substitutes for the program providing a request procedure. The stub accepts and forwards the request to the remote procedure. Once the remote procedure completes the request, it returns the results to the stub which in turn passes them to the request procedure.
```

Comment:
This paragraph seems to summarize the content of the following procedural steps in advance of them, which makes one or the other somewhat redundant. Maybe a more focused description of the stub specifically would make more sense in this highlight box:

**A stub is a small program routine substituting a longer program, allowing a machine to behave as if a program on a remote machine was operating locally. The client has a stub which interfaces with the remote procedure, while the server has a stub to interface with the original request procedure.**